### PR TITLE
[Microsoft.Build] Fix Microsoft.Build.Execution.BuildManagerTest.BasicManualParallelBuilds

### PR DESCRIPTION
**Pull request by ludovic-henry**
The previous implementation of ProcessWrapper would launch its own threads to read on the process stdout and stderr. We can more simply use OutputDataReceived and ErrorDataReceived which will do the same under the hood. We use as well the Exited event to track the end of the Process.

### CODE REVIEW

**Comment by ludovic-henry** [comment]
@akoeplinger that fixes the Microsoft.Build crash

@monojenkins merge

**Comment by akoeplinger** [comment]
LGTM, thanks!

File: mcs/class/Microsoft.Build.Utilities/Microsoft.Build.Utilities/ProcessWrapper.cs [hunk 1]
```diff
@@ -101,10 +79,9 @@ protected override void Dispose (bool disposing)
 			if (!done)
 				((IAsyncOperation)this).Cancel ();
 
-			captureOutputThread = captureErrorThread = null;
-			endEventOut.Close ();
-			endEventErr.Close ();
-			endEventOut = endEventErr = null;
+			endEventOut.Dispose (disposing);
+			endEventErr.Dispose (disposing);
+			endEventExit.Dispose (disposing);
```
Comment (akoeplinger) [inline mcs/class/Microsoft.Build.Utilities/Microsoft.Build.Utilities/ProcessWrapper.cs]: actually, I spoke too soon: this fails to compile:

```
MCS     [net_4_x] Microsoft.Build.Utilities.v4.0.dll
Microsoft.Build.Utilities/ProcessWrapper.cs(82,16): error CS1540: Cannot access protected member `System.Threading.WaitHandle.Dispose(bool)' via a qualifier of type `System.Threading.ManualResetEvent'. The qualifier must be of type `Microsoft.Build.Utilities.ProcessWrapper' or derived from it
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Microsoft.Build.Utilities/ProcessWrapper.cs(82,16): error CS0122: `System.Threading.WaitHandle.Dispose(bool)' is inaccessible due to its protection level
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Microsoft.Build.Utilities/ProcessWrapper.cs(83,16): error CS1540: Cannot access protected member `System.Threading.WaitHandle.Dispose(bool)' via a qualifier of type `System.Threading.ManualResetEvent'. The qualifier must be of type `Microsoft.Build.Utilities.ProcessWrapper' or derived from it
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Microsoft.Build.Utilities/ProcessWrapper.cs(83,16): error CS0122: `System.Threading.WaitHandle.Dispose(bool)' is inaccessible due to its protection level
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Microsoft.Build.Utilities/ProcessWrapper.cs(84,17): error CS1540: Cannot access protected member `System.Threading.WaitHandle.Dispose(bool)' via a qualifier of type `System.Threading.ManualResetEvent'. The qualifier must be of type `Microsoft.Build.Utilities.ProcessWrapper' or derived from it
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Microsoft.Build.Utilities/ProcessWrapper.cs(84,17): error CS0122: `System.Threading.WaitHandle.Dispose(bool)' is inaccessible due to its protection level
./../../class/lib/net_4_x/mscorlib.dll (Location of the symbol related to previous error)
Compilation failed: 6 error(s), 0 warnings
```

**Comment by radical** [comment]
Does this fix a bug? Could you please include details about that too?

**Comment by ludovic-henry** [comment]
That fixes a test failure on mono master, see https://jenkins.mono-project.com/job/test-mono-mainline/label=debian-armel/3513/parsed_console/log_content.html#WARNING1

**Comment by radical** [comment]
Could you please explain what the issue was and how this commit fixes it, in the commit message? The current commit message seems to suggest that this is just making the code simpler.

**Comment by ludovic-henry** [comment]
@radical I completely forgot about monojenkins... sorry about that :/

**Pull request merged**